sdk/typescript/src/cryptography/mnemonics.ts

```diff
@@ -18,13 +18,17 @@ export function isValidHardenedPath(path: string): boolean {
 }
 
 /**
- * Parse and validate a path that is compliant to BIP-32 in form m/54'/784'/{account_index}'/{change_index}/{address_index}.
+ * Parse and validate a path that is compliant to BIP-32 in form m/54'/784'/{account_index}'/{change_index}/{address_index}
+ * for Secp256k1 and m/74'/784'/{account_index}'/{change_index}/{address_index} for Secp256r1.
+ *
  * Note that the purpose for Secp256k1 is registered as 54, to differentiate from Ed25519 with purpose 44.
  *
  * @param path path string (e.g. `m/54'/784'/0'/0/0`).
  */
 export function isValidBIP32Path(path: string): boolean {
-  if (!new RegExp("^m\\/54'\\/784'\\/[0-9]+'\\/[0-9]+\\/[0-9]+$").test(path)) {
+  if (
```

Comment:
Do we want this function to take in the keypair type so that we can vary the validation?

Reply:
That makes sense. I'll make a follow up PR with that.